### ISSUE TITLE
Aspeed fan speed on P8 system incorrect

### ISSUE THREAD

**Issue by mine260309** [openbmc/openbmc#3162]
Tried to enable aspeed pwm driver on P8 Palmetto system, and find that the fan speed is incorrect.
The expected fan speed is about 13000 RPM at PWM 255, but the `fanX_input` reads about 33000 RPM.

The related changes (testing purpose only):
```
      1 diff --git a/arch/arm/boot/dts/aspeed-bmc-opp-palmetto.dts b/arch/arm/boot/dts/aspeed-bmc-opp-palmetto.dts
      2 index 9e2bb80..1c5c336 100644
      3 --- a/arch/arm/boot/dts/aspeed-bmc-opp-palmetto.dts
      4 +++ b/arch/arm/boot/dts/aspeed-bmc-opp-palmetto.dts
      5 @@ -337,3 +337,41 @@
      6                 line-name = "BMC_TPM_INT_N";
      7         };
      8  };
      9 +
     10 +&pwm_tacho {
     11 +       status = "okay";
     12 +       pinctrl-names = "default";
     13 +       pinctrl-0 = <&pinctrl_pwm0_default &pinctrl_pwm1_default
     14 +               &pinctrl_pwm2_default &pinctrl_pwm3_default
     15 +               &pinctrl_pwm4_default &pinctrl_pwm5_default>;
     16 +
     17 +       fan@0 {
     18 +               reg = <0x00>;
     19 +               aspeed,fan-tach-ch = /bits/ 8 <0x00>;
     20 +       };
     21 +
     22 +       fan@1 {
     23 +               reg = <0x01>;
     24 +               aspeed,fan-tach-ch = /bits/ 8 <0x01>;
     25 +       };
     26 +
     27 +       fan@2 {
     28 +               reg = <0x02>;
     29 +               aspeed,fan-tach-ch = /bits/ 8 <0x02>;
     30 +       };
     31 +
     32 +       fan@3 {
     33 +               reg = <0x03>;
     34 +               aspeed,fan-tach-ch = /bits/ 8 <0x03>;
     35 +       };
     36 +
     37 +       fan@4 {
     38 +               reg = <0x04>;
     39 +               aspeed,fan-tach-ch = /bits/ 8 <0x04>;
     40 +       };
     41 +
     42 +       fan@5 {
     43 +               reg = <0x05>;
     44 +               aspeed,fan-tach-ch = /bits/ 8 <0x05>;
     45 +       };
     46 +};
```

**Comment by shenki** [comment]
A fix for this was merged into the openbmc kernel as 9224ca752a6d305a1066ef249fd53d54e6d59ba7.

This will be available when https://gerrit.openbmc-project.xyz/c/10799/ is merged.

**Comment by rfrandse** [comment]
<https://gerrit.openbmc-project.xyz/10799> kernel: Poleg pinctrl, clk fixes, KCS driver, pmbus fix
Fixes:  #3162 Aspeed fan speed on P8 system incorrect
